### PR TITLE
chore(weave): Add back hook to check for weave data

### PR DESCRIPTION
**Pull request by ericakdiaz**
## Description

This basically reverts the cleanup from https://github.com/wandb/weave/pull/2132. I believe that the limit issue should've been fixed in https://github.com/wandb/weave/pull/2420, so hopefully we should be able to check if a project has weave data more quickly.

## Testing

How was this PR tested?

This should have no effect since it's unused, will follow up in core with a change similar to this https://github.com/wandb/core/pull/20661/commits/3ca824472beff5023cd0c4926e1f95d9cbc63f1c and see how the performance is now.

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/context.tsx [hunk 1]
```diff
@@ -35,3 +36,44 @@ export const WFDataModelAutoProvider: FC<{
...
+  entity: string,
+  project: string
+) => {
+  const hasTraceServer = useHasTraceServerClientContext();
+  const objs = tsWFDataModelHooks.useRootObjectVersions(
+    entity,
+    project,
+    {},
+    1,
+    true,
```
Comment (ericakdiaz) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/context.tsx:54]: added metadata_only param

**Review by jamie-rasmussen — APPROVED** [review]
Needs lint

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/context.tsx [hunk 2]
```diff
@@ -35,3 +36,44 @@ export const WFDataModelAutoProvider: FC<{
...
+    {
+      skip: !hasTraceServer,
+    }
+  );
+
+  const calls = tsWFDataModelHooks.useCalls(
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/context.tsx:60]: you could optimize this by selecting only the `id` column for example as well

Comment (ericakdiaz) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/context.tsx:60]: done!

**Review by tssweeney — APPROVED** [review]

**Pull request merged**